core/message-tags-3.3.md

```diff
@@ -80,6 +89,24 @@ Individual tag keys MUST only be used a maximum of once per message. Clients
 receiving messages with more than one occurrence of a tag key SHOULD discard all
 but the final occurrence.
 
+### Empty tagged messages
+
+Servers MUST accept `PRIVMSG` and `NOTICE` events sent with client-only tags even if
+the message content parameter is omitted. These events MUST NOT be delivered to clients
+that haven't negotiated the message tags capability. If neither client-only tags nor a
+message parameter are included, servers MAY respond with a standard `ERR_NEEDMOREPARAMS`
+error.
+
+Clients that receive events without a message text MUST NOT display empty messages as-is
+in the message history unless they are using the attached tags according to their own
+specifications.
+
+### Size limit
+
+The size limit for message tags is increased from 512 to 4096 bytes, including the leading
```

Comment:
That would be in direct contravention of the recommendation in the spec.

> There is no guarantee that other clients will support or display the client-only tags they receive, so these tags should only be used to enhance messages with non-critical information. Messages should still make sense to clients with no support for tags.

If clients do that with their own prefixed tags, then they should expect it not to be widely adopted. I don't see it as a very effective incentive.